lib/finitomata.ex

```diff
@@ -488,16 +523,68 @@ defmodule Finitomata do
   end
 
   defp do_start_fsm(id, name, impl, payload) when is_atom(impl) do
+    {parent, payload} =
```

Comment:
Would you mind elaborating what happened here exactly, hard for me to infer from the code itself.

Reply:
We accept direct inferring of `parent` instead of default `self()`. UNder some circumstances (the process starts the task which in turn starts the FSM) it might be desired to explicitly set the parent pid/name.

Also, we accept _whatever_ shape of payload, it’s [not restricted to map](https://github.com/am-kantox/finitomata/blob/huge/flow/lib/finitomata.ex#L267-L271), or keyword, it’s a `term()`.

But if we receive either `map` or `keyword`, and it has a key `parent`, we pop it and use it as the `parent` in our state.